lib/_stream_readable.js

```diff
@@ -1140,30 +1140,37 @@ function endReadableNT(state, stream) {
   }
 }
 
-Readable.from = function(iterator, opts) {
+Readable.from = function(iterable, opts = {}) {
+  let iterator;
+  if (iterable && iterable[Symbol.asyncIterator])
+    iterator = iterable[Symbol.asyncIterator]();
+  else if (iterable && iterable[Symbol.iterator])
+    iterator = iterable[Symbol.iterator]();
+  else
+    throw new ERR_INVALID_ARG_TYPE('iterable', ['Iterable'], iterable);
+
+  if ('objectMode' in opts === false)
+    opts.objectMode = true;
```

Comment:
we shouldn't mutate objects passed by the user

Reply:
Sure.